src/executor/execute.rs

```diff
@@ -48,11 +48,19 @@ pub fn execute<T: 'static + Debug, U: Store<T> + StoreMut<T>>(
         Prepared::Create(schema) => storage
             .insert_schema(&schema)
             .map(|(storage, _)| (storage, Payload::Create)),
-        Prepared::Insert(table_name, row) => {
-            let (storage, key) = storage.generate_id(&table_name)?;
-            let (storage, row) = storage.insert_data(&key, row)?;
-
-            Ok((storage, Payload::Insert(row)))
+        Prepared::Insert(table_name, rows) => {
+            match rows.into_iter().try_fold(
+                (storage, Vec::<Row>::new()),
+                |(storage, mut rows), row| {
+                    let (storage, key) = storage.generate_id(&table_name)?;
+                    let (storage, row) = storage.insert_data(&key, row)?;
+                    rows.push(row);
+                    Ok((storage, rows))
+                },
+            ) {
+                Ok((storage, rows)) => Ok((storage, Payload::Insert(rows))),
```

Comment:
```rust
rows.into_iter().try_fold(...).map(|(storage, rows)| (storage, Payload::Insert(rows))
```
Changing value type is only required for `Ok`, then you can simply use `map` rather than `match`.